src/backend/distributed/commands/table.c

```diff
@@ -1629,8 +1676,36 @@ PostprocessAlterTableStmt(AlterTableStmt *alterTableStatement)
 						{
 							AttrNumber attnum = get_attnum(relationId,
 														   columnDefinition->colname);
-							Oid seqTypId = GetAttributeTypeOid(relationId, attnum);
-							EnsureSequenceTypeSupported(relationId, attnum, seqTypId);
+							Oid seqOid = GetSequenceOid(relationId, attnum);
+							if (seqOid != InvalidOid)
+							{
+								/*
+								 * We should make sure that the type of the column that uses
+								 * that sequence is supported
+								 */
+								Oid seqTypId = GetAttributeTypeOid(relationId, attnum);
+								EnsureSequenceTypeSupported(relationId, seqTypId);
+
+								/*
+								 * Alter the sequence's data type in the coordinator if needed.
+								 * A sequence's type is bigint by default and it doesn't change even if
+								 * it's used in an int column. We should change the type if needed,
+								 * and not allow future ALTER SEQUENCE ... TYPE ... commands for
+								 * sequences used as defaults in distributed tables
+								 */
+								AlterSequenceType(seqOid, seqTypId);
+
+								if (ShouldSyncTableMetadata(relationId) &&
+									ClusterHasKnownMetadataWorkers())
+								{
+									/* get sequence address */
+									ObjectAddress sequenceAddress = { 0 };
+									ObjectAddressSet(sequenceAddress, RelationRelationId,
+													 seqOid);
+									EnsureDependenciesExistOnAllNodes(&sequenceAddress);
+									MarkObjectDistributed(&sequenceAddress);
+								}
```

Comment:
This block is repeated twice, so might be nice to move it to a separate function.